### ISSUE TITLE
Dropdown aria-activedescendant issue

### ISSUE THREAD

**Issue by origami-z** [jpmorganchase/salt-ds#4412]
### Latest version

- [X] I have tested the latest version

### Description

The salt dropdown has aria-activedescendant issue report by the axe

> ARIA attributes must conform to valid values
> To solve this problem, you need to fix the following:
> Invalid ARIA attribute value: aria-activedescendant=":r4:"

### Steps to reproduce

Go to dropdown first default example, open the dropdown and select an item. Observe `aria-activedescendant` referring to invalid item (may need to repeat selection a few times to see this)
https://www.saltdesignsystem.com/salt/components/dropdown/examples

### Expected behavior

No axe error

### Package name(s)

Core (@salt-ds/core)

### Package version(s)

1.37.1

### Browser

- [X] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Microsoft Edge

### Operating system

- [X] macOS
- [ ] Windows
- [ ] Linux
- [ ] iOS
- [ ] Android

### Are you a JPMorgan Chase & Co. employee?

- [X] I am an employee of JPMorgan Chase & Co.

SO Question 94028

**Comment by joshwooding** [comment]
I cannot reproduce this

**Comment by joshwooding** [comment]
After testing, this seems to happen if you:

1. Visit https://www.saltdesignsystem.com/salt/components/dropdown/examples#default-single-select
2. Select the second item
3. Re-open and select the third item
